ports/stm32f4/common-hal/microcontroller/__init__.c

```diff
@@ -42,14 +42,30 @@
 // This routine should work even when interrupts are disabled. Used by OneWire
 // for precise timing.
 void common_hal_mcu_delay_us(uint32_t delay) {
-  //TODO: implement equivalent of mp_hal_delay_us(delay);
-  //this is fairly annoying in the STM32 HAL
+    // sys freq is always a multiple of 2MHz, so division here won't lose precision
+    const uint32_t ucount = HAL_RCC_GetSysClockFreq() / 2000000 * delay / 2;
+    for (uint32_t count = 0; ++count <= ucount;) {
```

Comment:
This delay could vary based on the optimization level, etc. The examples I see online use `DWT->CYCCNT` ,e.g https://github.com/keatis/dwt_delay/blob/master/dwt_delay.c